src/ucs/sys/sock.c

```diff
@@ -723,6 +723,20 @@ ucs_status_t ucs_sockaddr_copy(struct sockaddr *dst_addr,
     return UCS_OK;
 }
 
+void ucp_sockaddr_copy_always(struct sockaddr *dst_addr,
+                              const struct sockaddr *src_addr)
+{
+    ucs_status_t status;
+
+    if (!ucs_sockaddr_is_known_af(src_addr)) {
+        memset(dst_addr, 0, sizeof(*dst_addr));
+    } else {
+        status = ucs_sockaddr_copy(dst_addr, src_addr);
+        ucs_assert(status == UCS_OK);
+    }
```

Comment:
can be simplified:
```
status = ucs_sockaddr_copy(...)
if (status != UCS_OK) {
    memset(...);
}
```

Reply:
i need to avoid the printing of the error from ucs_sockaddr_sizeof (in case ucs_sockaddr_copy returns as status that isn't UCS_OK)

Reply:
is this case a real error? is yes - why it's a problem? otherwise - I think we can convert that error to debug/diag and handle it outside.

Reply:
from ucs_sockaddr_sizeof's perspective it's a real error so imo it should remain ucs_error. and it can happen if ucp_ep_query is called too early, before the rdmacm_id, for example, is bound to an address. 
but we can avoid printing this error from ucx by checking the address family first and if we set zeros, in case it's not ready yet, the user can handle this case as well.

Reply:
in this case we should return error, for example NOT_CONNECTED, also UCT has all requires callbacks to indicate the actual status, so UCP should be able to handle this as well even without erroneous call.

Reply:
I guess we cannot control addr value before it's bound, can it be not initialized and contain IP AF value by mistaken?

Reply:
but the query function didn't fail, it was called on an ep too early, i dont think it's an actual failure of the query function.
i think it would be better if the application would get a notification that the sockaddr isn't ready yet rather than getting a notice of a failure of an api function.

Reply:
from UCT API perspective if addr is querryed but not bound yet better to return status (since it's able to notify the status with CBs) which can be handled in UCP if we don't have actual state of CM lane to avoid err UCT call

Reply:
which uct callback are you referring to in this case? it's just a query function...

Reply:
connection establishment CBs which report ep status during wireup

Reply:
the query function can be called at any stage, after the wireup finished

Reply:
right, but before uct_ep has addr UCT should return err, is not the case you described?
> and it can happen if ucp_ep_query is called too early

Reply:
let's design the API to follow getsockname/getpeername
for example, https://man7.org/linux/man-pages/man2/getpeername.2.html return ENOTCONN if not connected yet